File: Codeforces/Codeforces Round #789/B_1_Tokitsukaze_and_Good_01_-String_easy_version.cpp
// Codeforces 1678B1  Tokitsukaze and Good 01-String (easy version)
// https://codeforces.com/contest/1678/problem/B1

#include "bits/stdc++.h"
using namespace std;


int solve() {
    int n; cin >> n;
    string s; cin >> s;
    int ans = 0;
    for (int i = 0; i < n; i += 2) {
        if (s[i] != s[i+1]) {
            ans++;
        }
    }
    return ans;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        cout << solve() << endl;
    }
}

File: Codeforces/Codeforces Round #789/B_1_Tokitsukaze_and_Good_01_-String_easy_version.py
# Codeforces 1678B1  Tokitsukaze and Good 01-String (easy version)
# https://codeforces.com/contest/1678/problem/B1

import sys
input = sys.stdin.readline
from array import array


def solve():
    n = int(input())
    s = input().rstrip()
    count = 0
    for i in range(0, len(s), 2):
        if s[i]!=s[i+1]:
            count += 1
    return count

for _ in range(int(input())):
    print(solve())